compaction/compaction.cc

```diff
@@ -826,6 +827,7 @@ class compaction {
         _estimated_droppable_tombstone_ratio = std::min(1.0, sum_of_estimated_droppable_tombstone_ratio / ssts->size());
 
         _compacting = std::move(ssts);
+        _reader_statistics.rows_merged_histogram.resize(_compacting->size()+1, 0);
```

Comment:
As noted before, we should let make_combined_reader (or perhaps its immediate callee) resize the vector, since it is the one who is intimately familiar with how it will be used. To be safe, it should only resize it upwards.